pkg/crd/init.go

```diff
@@ -313,14 +317,53 @@ func (c CRD) ToCustomResourceDefinition() (runtime.Object, error) {
 	if err != nil {
 		return nil, err
 	}
-	mapData["kind"] = "CustomResourceDefinition"
-	mapData["apiVersion"] = apiext.SchemeGroupVersion.String()
+	mapData["kind"] = CRDKind
+	mapData["apiVersion"] = apiextv1.SchemeGroupVersion.String()
 
 	return &unstructured.Unstructured{
 		Object: mapData,
 	}, unstructured.SetNestedField(mapData, false, "spec", "preserveUnknownFields")
 }
 
+func (c CRD) ToCustomResourceDefinitionV1Beta1() (*apiextv1beta1.CustomResourceDefinition, error) {
+	toConvertCRD, err := c.ToCustomResourceDefinition()
+	if err != nil {
+		return nil, err
+	}
+	if toConvertCRD == nil {
+		return nil, fmt.Errorf("cannot convert empty CRD runtime object to apiextensions v1beta1 CRD object")
+	}
+
+	unstructuredCRD, ok := toConvertCRD.(*unstructured.Unstructured)
+	if !ok {
+		return nil, fmt.Errorf("could not convert CRD runtime object to *unstructured.Unstructured")
+	}
+	var v1CRD *apiextv1.CustomResourceDefinition
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(unstructuredCRD.UnstructuredContent(), &v1CRD); err != nil {
+		return nil, err
+	}
+
+	internalCRD := &apiext.CustomResourceDefinition{}
+	if err := apiextv1.Convert_v1_CustomResourceDefinition_To_apiextensions_CustomResourceDefinition(v1CRD, internalCRD, nil); err != nil {
+		return nil, err
+	}
+	v1beta1CRD := &apiextv1beta1.CustomResourceDefinition{}
+	if err := apiextv1beta1.Convert_apiextensions_CustomResourceDefinition_To_v1beta1_CustomResourceDefinition(internalCRD, v1beta1CRD, nil); err != nil {
+		return nil, err
+	}
+
+	if &v1beta1CRD == nil {
+		return nil, fmt.Errorf("conversion resulted in a nil apiextensions v1beta1 CRD object")
+	}
+	// GVK is dropped during conversion, so we must add it.
+	v1beta1CRD.GetObjectKind().SetGroupVersionKind(schema.GroupVersionKind{
```

Comment:
Probably not needed since you are returning `apiextv1beta1.CustomResourceDefinition` but it should be fine

Reply:
I did need it since the second autogenerated function does not set the GVK again if it's empty.